agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -575,8 +585,20 @@ bool monitor_thread::update_sta_stats()
             std::chrono::duration_cast<std::chrono::milliseconds>(now - sta_stats.last_update_time);
         sta_stats.delta_ms         = float(time_span.count());
         sta_stats.last_update_time = now;
+
+        if (std::chrono::steady_clock::now() > awake_timeout()) {
+            // If we haven't finish to iterate on all stations, set the select timeout to minimal
+            // value (1 msec) temporarily so the select will not be stuck on full select timeout
+            // and the thread will be able to finish this operation quickly.
+            set_select_timeout(1, false);
+            return true;
+        }
```

Comment:
If I'm not mistaken basically here you are lowering the  after_select cycle to 75% .
All other task will be polled more frequently , CPU will increase and I'm not sure we want that.

Why not just continue to other tasks when `awake_timeout() `is reaching 75% and keep the cycle as is.
You could also move the update_sta_stats() to the end of the after_select that way you will make sure 
after select will be occupied with tasks only   75% of the time.

Reply:
We talked about this on the phone. Resolving for now.